newIDE/app/src/ResourcesList/index.js

```diff
@@ -79,6 +83,47 @@ export default class ResourcesList extends React.Component<Props, State> {
     this.props.onDeleteResource(resource);
   };
 
+  _scanForNewResources = () => {
+    const project = this.props.project;
+    const resourcesManager = project.getResourcesManager();
+    console.log('Scanning the project folder for new resources...');
+    const projectPath = path.dirname(project.getProjectFile());
+
+    const getDirectories = (src, callback) => {
+      glob(src + '/**/*.{png,jpg,jpeg,PNG,JPG,JPEG}', callback);
+    };
+    getDirectories(projectPath, (err, res) => {
+      if (err) {
+        console.log('Error loading ', err);
+      } else {
+        res.forEach(pathFound => {
+          const fileName = path.relative(projectPath, pathFound);
+          if (!resourcesManager.hasResource(fileName)) {
+            const imageResource = new gd.ImageResource();
+            imageResource.setFile(fileName);
+            imageResource.setName(fileName);
+            resourcesManager.addResource(imageResource);
+            imageResource.delete();
+            console.log(fileName + ' added to Project!');
+          }
+        });
+      }
+      this.forceUpdate();
+    });
+  };
+
+  _removeAllUnusedImages = () => {
+    const project = this.props.project;
+    gd.ProjectResourcesAdder
+      .getAllUselessImages(project)
+      .toJSArray()
+      .forEach((item, i) => {
```

Comment:
You don't need `i` here.
Also `item` is a bit too generic and don't convey what the variable is to the reader. Prefer something like `resourceName` or `imageName`:

```
.forEach(item => {
  console.info(`Removing unused image resource: ${item}`);
});
```

Prefer using "image resource" than "Image File", because what getAllUselessImages return is a list of resource names, not a list of resource files :)
(as previously, I've used `console.info` and string interpolation for clarity)

Reply:
ah yes you are right. It gives better formatting. Not sure why I forgot the "i" there. I was thinking of adding a number against each, but then decided against it